src/blockcrawler/evm/rpc.py

```diff
@@ -195,6 +195,77 @@ async def get_block(self, block_num: HexInt, full_transactions: bool = False) ->
         )
         return block
 
+    async def get_block_by_timestamp(self, timestamp: int) -> int:
+        """Get block number by timestamp.
```

Comment:
Please elaborate on what happens when there is an exact match and when there is no exact match.